File: Python/41-51/e_43.py
# Escribe en python una función maximo(L) que tome una lista de enteros como parámetro y devuelva el mayor valor. La idea es usar la función max.


def maximo(L):
    if not L:
        # Manejar el caso de una lista vacía
        return None
    else:
        # Utilizar la función max para obtener el valor máximo de la lista
        return max(L)

# Ejemplo de uso
lista_numeros = [10, 5, 8, 15, 3]
resultado = maximo(lista_numeros)
print(resultado)


# En este caso, la función maximo simplemente devuelve el resultado de aplicar la función max a la lista L.
# Si la lista está vacía, la función devuelve None. Puedes probar la función con diferentes listas de enteros para encontrar el valor máximo.

File: Java/src/e41/e_43.java
package e41;

import java.util.Collections;
import java.util.List;

public class e_43 {
    public static Integer maximo(List<Integer> L) {
        if (L.isEmpty()) {
            // Manejar el caso de una lista vacía
            return null;
        } else {
            // Utilizar Collections.max para obtener el valor máximo de la lista
            return Collections.max(L);
        }
    }

    public static void main(String[] args) {
        // Ejemplo de uso
        List<Integer> listaNumeros = List.of(10, 5, 8, 15, 3);
        Integer resultado = maximo(listaNumeros);
        System.out.println(resultado);
    }
}

// En este ejemplo, la función maximo toma una lista de enteros y utiliza
// Collections.max para encontrar el valor máximo.
// Si la lista está vacía, la función devuelve null. Puedes probar la función
// con diferentes listas de enteros para encontrar el valor máximo.